### PR TITLE
fix(client_encrypt): make it work when server encryption is disabled

### PR DESCRIPTION
**Pull request by vponomaryov**
Fixes: https://github.com/scylladb/scylla-cluster-tests/issues/8846

### Testing

- [scylla-staging/valerii/vp-longevity-aws-custom-d2-workload1-multidc#14](https://argus.scylladb.com/test/2c0a3aad-2dc8-4718-923e-af6271930f70/runs?additionalRuns[]=19520099-2f6f-41eb-999e-ebf170baa282)
- :green_circle: [pr-provision-test with client encrypt](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dimakr/job/sct-pr-provision-test/79/)
- :green_circle: [pr-provision-test with server encrypt](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dimakr/job/sct-pr-provision-test/80/)
- :green_circle: [pr-provision-test with server+client encrypt](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dimakr/job/sct-pr-provision-test/82/)

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
It must go to any branch where https://github.com/scylladb/scylla-cluster-tests/pull/7822 went.

**Comment by vponomaryov** [comment]
Error we get if we enabled only client encryption:

```
2024-09-24T16:03:42.214+00:00 long-custom-d2-wrkld1-small-dev-db-node-bc3eace1-1      \
    !ERR | scylla[6849]:  [shard 0:main] init - Startup failed: \
    std::_Nested_exception<std::runtime_error> (Could not read certificate file /etc/scylla/ssl_conf/client-facing.crt): \
    std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: \
    open failed: No such file or directory [/etc/scylla/ssl_conf/client-facing.crt])
```

File: sdcm/cluster.py [hunk 1]
```diff
@@ -4692,7 +4692,7 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
                 datacenters = self.datacenter  # pylint: disable=no-member
             SnitchConfig(node=node, datacenters=datacenters).apply()
 
-        if self.params.get('server_encrypt'):
+        if self.params.get('client_encrypt'):
```
Comment (dimakr) [inline sdcm/cluster.py:4695]: Change of this condition will break server encryption, as certificates for dbnode-to-dbnode encrypted communication will be created only if 'client_encrypt' is enabled.

Comment (dimakr) [reply sdcm/cluster.py:4695]: The problem is indeed in this condition, but on lines:
```
            # Create client facing node certificate, for client-to-node communication
            node.create_node_certificate(
                node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_CERT, node.ssl_conf_dir / TLSAssets.DB_CLIENT_FACING_KEY)
```
This is the server certificate, which is facing to client during client-to-dbnode communication.
It should be created in a separate `if self.params.get('client_encrypt'):` condition in this node setup method.

Comment (vponomaryov) [reply sdcm/cluster.py:4695]: Both parts get used in the client encrypt-only setup.
See failed test run here: [scylla-staging/valerii/vp-longevity-aws-custom-d2-workload1-multidc#13](https://argus.scylladb.com/test/2c0a3aad-2dc8-4718-923e-af6271930f70/runs?additionalRuns[]=7ab9b6e8-b782-4786-9c5b-30e831d75440)

Comment (fruch) [reply sdcm/cluster.py:4695]: @dimakr please take over this PR, 

make sure we have a test run to test client / server / client+server

Comment (dimakr) [reply sdcm/cluster.py]: Done. The fix turns out to be trivial - as during initializing DB cluster there is a peer check when DB nodes talk to each other for a moment, the server certificates should also be created, not just client-facing ones.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**